winch/codegen/src/codegen.rs

```diff
@@ -63,13 +63,12 @@ where
         &mut self,
         body: &mut BinaryReader<'a>,
         validator: FuncValidator<ValidatorResources>,
-    ) -> Result<Vec<String>> {
+    ) -> Result<()> {
         self.emit_start()
             .and(self.emit_body(body, validator))
```

Comment:
pre-existing but do we want to use `.and_then(|| self.emit_body(...))` and similar below as well? Otherwise if there's an error emitting the prologue we still emit the rest, then eventually return the error.

Reply:
Fixed!